### PR TITLE
[WEB-147] Next 6  upgrade & migration

### PR DESCRIPTION
**Pull request by UncleClapton**
## Core Changes
**\<Page />** - Our page component has been effectively split into 3 separate components

| Component | Purpose / Scope |
|----|----|
|`/pages/_app.js`|First-priority initialization tasks such as configuring localforage and FontAwesome. Provides the redux store from withRedux to the rest of the app. Is limited to basic setup since redux cannot be properly bound to this component.|
|`/components/AppLayout.js`|Controls application layout and performs the bulk of session initialization. user session verification, current user data collection, and authenticated page redirects are all managed here|
|`/components/Page.js`|The old page component has been rewritten into a basic HoC which stores info used by `<AppLayout />`, and also handles redux bindings in the same fashion as before|

**Upgrade to Babel 7** - Next 6 is forcing us to upgrade to Babel 7, which changes how we bring in new official babel transforms. When adding a new **official** babel transform, make sure it comes from the `@babel` namespace, and the version is explicity set to the current beta version of `@babel/core`. (currently 7.0.0-beta.42)

**Downgrade to Webpack 3** - We upgraded our Webpack instance to `webpack@4.x`, however because Next builds our app, the webpack config and the actual running version of webpack is still `webpack@3.x`. I downgraded us to webpack 3 so we don't run into any issues using the library in our config. 

**Class Properties** - We now have the class properties transform \o/. Welcome to the modern age!

**User session** - The user's id is now stored in the authentication global state object, and will be the preferred way to access the current user's id. Retrieving the ID through cookies should be discouraged as it is incompatible with SSR.

The user object for the current user is automatically fetched on the server when verifying the session. It is now ready for use in the initial draw. 

## Other Changes

**Page layout persistence** - Previously, since the page layout was a wrapper on top of the page, the entire DOM tree was remounted on every page change. Next 6 allows us to couple our site layout to the app itself, so this is no longer an issue. Anything mounted above the "Component" prop passed to `<AppLayout />` from `_app.js` will persist across page changes. 

**Error pages** - The error page gets passed to `<AppLayout />` like any other page, so our site banner can persist through error pages now. So that's neat.

**Banner subnav behavior** - Clicking on a open subnav header will now close the subnav. This change was made in the interest of keeping to user expectations of a responsive UX (insert more CS mumbo jumbo here. it makes it look not broken even though it wasn't in the first place).

**Redux connect helper** - The function we used in the original Page HoC has been moved to it's own helper. *All* components can now bind store actions through the string array syntax we have been using for pages.

## Known issues
`babel-loader` has some weird error that is currently pending a fix. The solution was to go back to `babel-loader@7.1.4`. This works fine with Babel 7, but has a peer dependency of `babel-core@6`. This just means we will see a incorrect peerDep warning in console, but it's not a big issue.

### CODE REVIEW

File: components/Nav.js [hunk 1]
```diff
@@ -148,6 +146,30 @@ const navItems = [
...
+
+  /***************************************************************************\
+    Private Methods
+  \***************************************************************************/
+
+  _handleSubnavChange = ({ target }) => {
+    const { openSubNav } = this.state
+    const { id } = target
+
+    this.setState({ openSubNav: openSubNav === id ? '' : id })
```
Comment (trezy) [inline components/Nav.js]: `_handleSubnavChange` vs `this.state.openSubNav`... Consistency, man!

File: package.json [hunk 2]
```diff
@@ -108,8 +110,8 @@
     "sw-precache-webpack-plugin": "^0.11.5",
     "uglifyjs-webpack-plugin": "^1.2.5",
     "uuid": "^3.2.1",
-    "webpack": "^4.8.3",
-    "webpack-bundle-analyzer": "^2.12.0",
+    "webpack": "3.10.0",
```
Comment (trezy) [inline package.json]: We're downgrading our Webpack? That seems off...

Comment (UncleClapton) [reply package.json]: Forgot to mention this. Adding explainer to the main description.

File: components/Nav.js (hunk 1 above)
Comment (UncleClapton) [reply components/Nav.js]: whoops XD

**Pull request merged**